### PR TITLE
TCP Client and Server Sending Data 

### LINKED ISSUE #35: Complete TCP Data Transmission for Pilot Data
Enhance the basic TCP client implementation to include sending flight data from the X-Plane plugin to a server. The focus is on completing the TCP connection and adding the code to send the extracted flight data (e.g., altitude, airspeed) to the test server.

Developer Notes:

- [ ] Use the existing TCP client code to establish a connection.
- [ ] Implement functionality to send dummy flight data to simulate transmission.
- [ ] This will act as a prototype for future transmission to the iMotions platform once the server is ready.
- [ ] Update the code to handle potential transmission errors gracefully.

Acceptance Criteria:

The TCP client successfully transmits test data (e.g., hardcoded values like altitude or airspeed) to a test server.
Verify the data reaches the server and is received correctly.
The code should be clean, well-documented, and include transmission logic.
Add comments explaining how the data will be formatted and prepared for future integration with the iMotions API.

Testing Notes:

Verify that the TCP client can send flight data to the server without errors.
Document the testing process in the README.

### PR DESCRIPTION
**Pull request by carlyrhoover**
### Fixes #35 

---

### **What was changed?**

TCP.cpp was split into two .cpp files: TCPClient.cpp and TCPServer.cpp, both were modified from their original state when in TCP.cpp.
---

### **Why was it changed?**

Client and server were split into two files in order to run them concurrently to send hard-coded test data. Compile and run each file in a separate terminal and they should connect. 

---

### **How was it changed?**

---

### **Screenshots (if applicable):**
Attached is a screenshot of what the server looks like waiting, and then the client sending the data. 
serverwaiting
clientoutput

### CODE REVIEW

**Comment by kate-holdener** [comment]
@carlyhoover you'll want to wait for PR #33 to get merged, and then create a branch from main and add your code changes to that branch.

**Comment by carlyrhoover** [comment]
PR #33 has been fixed! Thanks!

File: src/TCP.cpp
Comment (aelnourSLU) [inline src/TCP.cpp:1]: I assume that this file formerly held the contents of both the TCP server and TCP client?

File: include/plugin.h
Comment (aelnourSLU) [inline include/plugin.h:1]: No issues with it, but I'd like to confirm that the plan is for us to externalize header includes going forward. If so, let's get this merged ASAP

**Review by aelnourSLU — COMMENTED** [review]
Overall looks good; after implementing suggested changes, I think it should be good to merge. We'll have a good opportunity to stress-test the TCP additions for our discussed dummy test server, for testing our iMotions data transmission functions.

File: tests/plugin_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#include <gtest/gtest.h>
+#include "plugin.h"  // You'll need to create this header file
```
Comment (hrmeiners) [inline tests/plugin_tests.cpp:2]: Should this be commented out or even pushed? I am not sure where we are going to implement testing or if we are going to change the testing format or not? I am very confused here and we should confirm this with @harshitha-thota

**Review by hrmeiners — CHANGES_REQUESTED** [review]

File: tests/plugin_tests.cpp (hunk 1 above)
Comment (hrmeiners) [reply tests/plugin_tests.cpp:2]: I have the same confusion with the CMAKE File up top

**Comment by kate-holdener** [comment]
@carlyhoover please make sure to start each sprint with a new branch that originates with the latest code in main (create your branch from main, and then make your code changes). Your PR has files that should not be there, probably because you started from some old branch you had.

File: CMakeLists.txt
Comment (harshitha-thota) [inline CMakeLists.txt:1]: Hello Carly, please comment the testing part.

File: tests/plugin_tests.cpp (hunk 1 above)
Comment (harshitha-thota) [reply tests/plugin_tests.cpp:2]: Yes I have a new file this should be removed.

File: src/hello-world.cpp
[the remark this answers is not in the document]
Comment (harshitha-thota) [reply src/hello-world.cpp:1]: Yes, we can remove this file. As we are working on the plugin that does not include a basic hello-world.

File: src/TCPServer.cpp
Comment (harshitha-thota) [inline src/TCPServer.cpp:1]: Change the Port number to 8089. As we will connect to iMotions platform to this port number.

File: CMakeCache.txt
Comment (harshitha-thota) [inline CMakeCache.txt:1]: I think this file can be removed

File: CMakeFiles/3.31.0-rc3/CMakeSystem.cmake
Comment (harshitha-thota) [inline CMakeFiles/3.31.0-rc3/CMakeSystem.cmake:1]: CMakeFiles folder can be removed

**Review by harshitha-thota — APPROVED** [review]

**Pull request merged**